src/functions/getAddresses.ts

```diff
@@ -0,0 +1,231 @@
+import bitwise from 'bitwise';
+import { Byte, UInt4 } from 'bitwise/types';
+import {
+  ADDR_STR_LEN,
+  decResLengths,
+  EXTERNAL,
+  getFwVersionConst,
+} from '../constants';
+import { decryptResponse, encryptRequest, request } from '../shared/functions';
+import {
+  validateFwVersion,
+  validateIsUInt4,
+  validateNAddresses,
+  validateSharedSecret,
+  validateStartPath,
+  validateUrl,
+  validateWallet,
+} from '../shared/validators';
+import { isValidAssetPath } from '../util';
+
+/**
+ * `getAddresses` takes a starting path and a number to get the addresses associated with the
+ * active wallet.
+ * @category Lattice
+ * @returns An array of addresses.
+ */
+export async function getAddresses ({
+  startPath,
+  n,
+  flag: _flag,
+  client,
+}: GetAddressesRequestFunctionParams): Promise<Buffer[]> {
+  const { url, fwVersion, wallet, sharedSecret, flag } =
+    validateGetAddressesRequest({
+      startPath,
+      n,
+      flag: _flag,
+      url: client.url,
+      fwVersion: client.fwVersion,
+      wallet: client.getActiveWallet(),
+      sharedSecret: client.sharedSecret,
+    });
+
+  const payload = encodeGetAddressesRequest({
+    startPath,
+    n,
+    flag,
+    fwVersion,
+    wallet,
+  });
+
+  const encryptedPayload = encryptGetAddressesRequest({
+    payload,
+    sharedSecret,
+  });
+
+  const encryptedResponse = await requestGetAddresses(encryptedPayload, url);
+
+  const { decryptedData, newEphemeralPub } = decryptGetAddressesResponse(
+    encryptedResponse,
+    sharedSecret,
+  );
+
+  client.ephemeralPub = newEphemeralPub;
+
+  const data = decodeGetAddresses(decryptedData, flag);
+
+  return data;
+}
+
+export const validateGetAddressesRequest = ({
+  startPath,
+  n,
+  flag,
+  url,
+  fwVersion,
+  wallet,
+  sharedSecret,
+}: ValidateGetAddressesRequestParams) => {
+  validateStartPath(startPath);
+  validateNAddresses(n);
+  const validFlag = validateIsUInt4(flag);
```

Comment:
Seems more like a coerce function than a validate one (which I wouldn't expect to return anything) but idk

Reply:
What about the 3 under it? Not a big deal, I just think it's weird to name functions with return values "validation" functions